src/Chapter3.hs

```diff
@@ -480,6 +516,37 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+
+type CastleName = String
+type Walls = Int
+newtype House = House Int
+  deriving (Show)
+type Houses = [House]
+
+data MainBuilding = Church | Library
+  deriving (Show)
+
+data City = CityWithCastle CastleName Walls MainBuilding Houses
+  | OpenCity MainBuilding Houses
+  deriving (Show)
+
+getHousePeople :: House -> Int
+getHousePeople (House x) = x
+
+buildCastle :: City -> CastleName -> City
+buildCastle (CityWithCastle _ walls mainBuilding houses) castleName = CityWithCastle castleName walls mainBuilding houses
```

Comment:
I think that here pattern matching is not exhaustive. You need also to look into the case with the `OpenCity` 🙂